main.tf

```diff
@@ -37,6 +37,12 @@ locals {
   project_bucket_name = "${var.bucket_name != "" ? var.bucket_name : format("%s-state", var.name)}"
   create_bucket       = "${var.bucket_project != "" ? "true" : "false"}"
   gsuite_group        = "${var.group_name != "" || var.create_group}"
+  app_engine_enabled  = "${length(keys(var.app_engine)) > 0 ? true : false}"
+
+  app_engine_config = {
+    enabled  = "${list(var.app_engine)}"
```

Comment:
It gets the length of keys in app_engine which is a list

Reply:
But then I'm confused about how line 86 works? https://github.com/terraform-google-modules/terraform-google-project-factory/pull/4/files/a51eb1f65e51143af5b7dff92d9ce2fa268cfb38#diff-7a370d8342e7203b805911c92454f0f4R86

Reply:
local.app_engine_config is a map of lists with keys `enabled` and `disabled`. Line 86 is referencing that list by key depending if local.app_engine_enabled is true or false. If local.app_engine_enabled is true, it sets `app_engine` to the `enabled` list which is the users config, otherwise it uses the `disabled` list which is empty. It is a dirty hack, but the only way I could see to pass a completely empty list to `app_engine` and still be able to accept the `app_engine` variable as a map in our module. Otherwise, it would set `app_engine = [{}]` which isnt an empty list and tells the provider to setill enable app_engine.